### DISCUSSION TITLE
ConBee II "concurrency" issues

### DISCUSSION THREAD

**Discussion by vlinder** [Koenkk/zigbee2mqtt#13100] [Devices]
I have a ConBee II that works pretty well overall, but there is a use case I have issues with.

I'm using homebridge where I have made a scene that basically turns off 21 different devices. I also have an IKEA shortcut button that is used to trigger this scene.

My problem is that only a few of the devices actually gets turned off when I trigger the scene, around 10 or so. There seems to be some sort of concurrency issue. After about a minute there appears a lot of error messages in the zigbee2mqtt logs. Another thing that often happens at the same time is that some devices no longer have the correct OnOff status shown, especially in homebridge but also some times in zigbee2mqtt. Feels like reporting messages gets discarded sometimes or something.

What are the recommended settings for the deconz conbee II usb serial adapter?
* concurrency?
* delay?
* RTS / CTS?
* anything else?

I have tried playing around a bit with the settings above but haven't been able to solve the issues..

EDIT:
Just updated my firmware to the latest 2022 release ([deCONZ_ConBeeII_0x26780700.bin.GCF](https://deconz.dresden-elektronik.de/deconz-firmware/deCONZ_ConBeeII_0x26780700.bin.GCF)) and still having issues. Having set the concurrency down to 1 and delay to 100, it managed to turn off all the lights. But for some reason the reporting did not get back into zigbee2mqtt. This leaves me with 21 lights still showing as ON in the zigbee2mqtt dashboard, event though they are all actually turned off.

**Comment by Hellou1337** [comment]
this seems to be the same problem:
https://community.home-assistant.io/t/light-groups-malfunction-bulbs-toggle-differently-gif-attached/355283

would be great if someone posted a solution

putting the lights to groups seems to be a workaroud for now

**Reply by vlinder** [reply]
Seems like the issue is with the Conbee II.

I think it just runs out of memory to queue messages that haven't been sent yet.

Someone mentioned that using the Deconz software solved it for them. They know the limitations of their own device and are probably caching requests on the host machine, sending them in a pace the Conbee II can handle.

A lot of guessing here, but it should probably be solvable in software.

Maybe I could dig up some old serial port debugging software to see the differences in how different software communicates with the Conbee II..

**Reply by kalbfuss** [reply]
I am observing similar limitations with the RaspBee II hat. Lights do not switch reliably when individually included in scenes in Home Assistant. Grouping the lights in z2m and using the groups in Home Assistant is a lot more reliable, albeit not perfect.
Is it possible to tweak settings for the Conbee II/RaspBee 2? How is the performance with other Zigbee coordinators?

**Reply by System67** [reply]
Same problem here, Conbee II with Openhab 3  and native Openhab Deconz binding and it wasn't  possible to set 17 lights in one go. Therefore I used a 1 sec delay after every 3 lights to get it stable. Yesterday I moved over to z2m and Sonoff 3.0 E coordinator (using groups and scenes) and now it's nearly instant on/off.
